vald/evm/evm.go

```diff
@@ -466,69 +464,50 @@ func getLatestFinalizedBlockNumber(client rpc.Client, confHeight uint64) (*big.I
 	return big.NewInt(int64(blockNumber - confHeight + 1)), nil
 }
 
-func (mgr Mgr) validate(client rpc.Client, txID common.Hash, confHeight uint64, validateTx func(tx *geth.Transaction, txReceipt *geth.Receipt) bool) bool {
-	tx, _, err := client.TransactionByHash(context.Background(), txID)
-	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "get transaction by hash call failed").Error())
-		return false
+func (mgr Mgr) getTxReceiptIfFinalized(chain nexus.ChainName, txID common.Hash, confHeight uint64) (*geth.Receipt, error) {
+	client, ok := mgr.rpcs[strings.ToLower(chain.String())]
+	if !ok {
+		return nil, fmt.Errorf("rpc client not found for chain %s", chain.String())
 	}
 
 	txReceipt, err := client.TransactionReceipt(context.Background(), txID)
-	if err != nil {
-		mgr.logger.Debug(sdkerrors.Wrap(err, "get transaction receipt call failed").Error())
-		return false
+	if err == ethereum.NotFound {
+		return nil, nil
 	}
-
-	if !isTxSuccessful(txReceipt) {
```

Comment:
Why remove this check for successful execution?

Reply:
cuz it's useless. Unsuccessful tx receipt cannot have any logs in it.